server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -467,18 +468,13 @@ protected Node(final Environment initialEnvironment,
                 .collect(Collectors.toUnmodifiableMap(
                     plugin -> plugin.getClass().getSimpleName(),
                     plugin -> plugin.getSystemIndexDescriptors(settings)));
-            SystemIndexDescriptor.checkForOverlappingPatterns(systemIndexDescriptorMap);
```

Comment:
This call to `checkForOverlappingPatterns` got dropped in the change - did you mean to move it into the `SystemIndices` constructor?